File: eid-ro/src/eID.RO.Contracts/Results/UserIdentifier.cs
using eID.RO.Contracts.Enums;

namespace eID.RO.Contracts.Results;

public interface UserIdentifier
{
    string Uid { get; }
    IdentifierType UidType { get; }
    string Name { get; set; }
}

[Serializable]
public class UserIdentifierData : UserIdentifier
{
    public string Uid { get; set; } = string.Empty;
    public IdentifierType UidType { get; set; }
    public string Name { get; set; } = string.Empty;
}

public interface AuthorizerIdentifier : UserIdentifier
{
    bool IsIssuer { get; }
}

[Serializable]
public class AuthorizerIdentifierData : AuthorizerIdentifier
{
    public string Uid { get; set; } = string.Empty;
    public IdentifierType UidType { get; set; }
    public string Name { get; set; } = string.Empty;
    public bool IsIssuer { get; set; }
}

File: eid-mobile-ios/eID-Mobile-iOS/02_Data/Entities/EmpowermentsRegister/Common/UserIdentifier.swift
//
//  UserIdentifier.swift
//  eID-Mobile-iOS
//
//  Created by Sunay Gyultekin on 8.01.24.
//

import Foundation


struct UserIdentifier: Codable, Hashable {
    // MARK: - Properties
    var uid: String?
    var uidType: IdentifierType?
    var name: String?
    
    var displayValue: String {
        let displayName = name?.isEmpty == true ? "" : " - \(name ?? "")"
        return "\(uidType?.title.localized() ?? ""): \(uid ?? "")\(displayName)"
    }
}
